app/src/i18n/utils.ts

```diff
@@ -0,0 +1,37 @@
+import { ui, fallbackLang, languages } from './ui'
+
+export const getLangFromLocalStorage = () => {
+  const lang = localStorage.getItem('lng')
+
+  if (lang && Object.keys(languages).includes(lang)) {
+    return lang
+  }
+
+  return fallbackLang
+}
+
+export const getLangFromWindowUrl = () => {
```

Comment:
Unit test this please.
You probably will need to mock `window.location.pathname`.